assets/scripts/utils/fontFacesWatcher.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Watch font faces used in our website and trigger a custom event `fontsLoaded` when all font faces are loaded.
+ *
+ *  @param {Object[]} fonts - List of fonts to watch
+ *  @param {string} fonts[].fontFamily - The family name used is our CSS to identify the font.
+ *  @param {string} fonts[].style - The family style used is our CSS by the font.
+ *  @param {string} fonts[].weight - The family weight used is our CSS by the font.
+ *  @param {boolean} isDebug - Allow console.log inside the function.
+ *
+ *  @return {Promise} Promise that helps us execute code when all font faces are loaded.
+ */
+export function fontFacesWatcher(fonts, isDebug = false) {
```

Comment:
Suggestions:

1. Add file-level block comment.
2. Define custom type for block comment tags.
3. Improve function block comment.
4. Rename `fontFamily` to `family`.
5. Rename `isDebug` to `debug`.

```suggestion
/**
 * Font Faces Watcher
 *
 * Utility for detecting when font faces are loaded
 * and dispatching a custom event.
 */

/**
 * @typedef {Object} FontFaceReference
 *
 * @property {string} family - The name used to identify the font in our CSS.
 * @property {string} style  - The style used by the font in our CSS.
 * @property {string} weight - The weight used by the font in our CSS.
 */

/**
 * Watches for the given font faces and triggers a custom event `fontsloaded`
 * when all font faces are loaded.
 *
 * @param {FontFaceReference[]} fonts - List of fonts to watch
 * @param {boolean}             debug - Log details to the console.
 *
 * @return {Promise}
 */
export function fontFacesWatcher(fonts, debug = false) {
```